cumulusci/robotframework/Salesforce.py

```diff
@@ -0,0 +1,477 @@
+        """ Validates that a field in the record header does not have a value.
+            NOTE: Use other keywords for non-string value types
+        """
+        locator = locators['record']['header']['field_value'].format(label)
+        self._call_selenium('_header_field_value_should_not_exist', True, locator)
+
+    def header_field_should_have_link(self, label):
+        """ Validates that a field in the record header has a link as its value """
+        locator = locators['record']['header']['field_value_link'].format(label)
+        self._call_selenium('_header_field_value_should_exist', True, locator)
+    
+    def header_field_should_not_have_link(self, label):
+        """ Validates that a field in the record header does not have a link as its value """
+        locator = locators['record']['header']['field_value_link'].format(label)
+        self._call_selenium('_header_field_value_should_not_exist', True, locator)
+
+    def header_field_should_be_checked(self, label):
+        """ Validates that a checkbox field in the record header is checked """
+        locator = locators['record']['header']['field_value_checked'].format(label)
+        self._call_selenium('_header_field_value_should_exist', True, locator)
+
+    def header_field_should_be_unchecked(self, label):
+        """ Validates that a checkbox field in the record header is unchecked """
+        locator = locators['record']['header']['field_value_unchecked'].format(label)
+        self._call_selenium('_header_field_value_should_exist', True, locator)
+
+    def _header_field_value_should_exist(self, locator):
+        self.selenium.page_should_contain_element(locator)
+    
+    def _header_field_value_should_not_exist(self, locator):
+        self.selenium.page_should_not_contain_element(locator)
+
+    def open_app_launcher(self):
+        """ EXPERIMENTAL!!! """
+        locator = locators['app_launcher']['button']
+        self._call_selenium('_open_app_launcher', True, locator)
+
+    def _open_app_launcher(self, locator):
+        self.builtin.log('Hovering over App Launcher button')
+        self.selenium.mouse_over(locator)
+        self.builtin.log('Clicking App Launcher button')
+        self.selenium.get_webelement(locator).click()
+        self.builtin.log('Waiting for modal to open')
+        self.wait_until_modal_is_open()
+
+    def populate_field(self, name, value):
+        locator = locators['object']['field'].format(name)
+        self._call_selenium('_populate_field', True, locator, value)
+
+    def populate_lookup_field(self, name, value):
+        self.populate_field(name, value)
+        locator = locators['object']['field_lookup_value'].format(value)
+        self._call_selenium('_populate_lookup_field', True, locator)
+    
+    def _populate_lookup_field(self, locator):
+        self.selenium.set_focus_to_element(locator)
+        self.selenium.get_webelement(locator).click()
+        time.sleep(.5)
```

Comment:
can we pull out `SLEEP_SHORT=0.5` and `SLEEP_LONG=5` as module constants and use in all method?

Reply:
Looking through this, we're trying to keep these adjusted based on the performance of the individual method.  There are too many possible sleep values already to justify storing them all as separate variables.  Since the test is tied to the keyword's method, an unstable test tells us we might want to adjust that individual sleep.

Also note, sleeping is an interim hack until we can figure out how to better tap into component states via $A.